src/librustc_typeck/check/method/confirm.rs

```diff
@@ -133,10 +132,10 @@ impl<'a, 'gcx, 'tcx> ConfirmContext<'a, 'gcx, 'tcx> {
             ty: fty,
             substs: all_substs
         };
-        // If this is an `&mut self` method, bias the receiver
-        // expression towards mutability (this will switch
-        // e.g. `Deref` to `DerefMut` in overloaded derefs and so on).
-        self.fixup_derefs_on_method_receiver_if_necessary(&callee);
+
+        if let Some(hir::MutMutable) = pick.autoref {
```

Comment:
Nit: Sad to lose the comment here. I realize it overlaps with `convert_lvalue_derefs_to_mutable`, but still seems like it will it easier for readers to follow.